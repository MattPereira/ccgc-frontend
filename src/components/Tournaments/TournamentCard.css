.TournamentCard-link {
    text-decoration: none;
    color: black;
}


.TournamentCard-card:hover { /* <Card /> that is also a <Link /> */
    text-decoration: none;
    color: #198754;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
  }

  .TournamentCard-crop {
    height: 350px;
    overflow: hidden;
  }

  @media (max-width: 1000px) {
    .TournamentCard-crop {
      height: 250px;
    }
  }

  .TournamentCard-crop img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
