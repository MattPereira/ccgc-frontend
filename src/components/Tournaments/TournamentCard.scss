.TournamentCard-crop {
    height: 300px;
    overflow: hidden;
  }

  @media (max-width: 768px) {
    .TournamentCard-crop {
      height: 250px;
    }
  }


  @media (max-width: 576px) {
    .TournamentCard-crop {
      height: 200px;
    }
  }

  .TournamentCard-crop img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    border-radius:4px 4px 0 0;
  }
