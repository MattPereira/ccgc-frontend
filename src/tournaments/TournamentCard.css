.TournamentCard-link {
    text-decoration: none;
    color: black;
}


.TournamentCard-card:hover { /* <Card /> that is also a <Link /> */
    text-decoration: none;
    color: #198754;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
  }

  .img-container {
    overflow: hidden;
    max-height: 400px;
    /* position: relative; */
  }
