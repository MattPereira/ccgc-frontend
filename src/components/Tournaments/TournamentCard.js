import React from "react";
import { Card } from "react-bootstrap";
import { Link } from "react-router-dom";
import "./TournamentCard.scss";

/** Tournament card component.
 *
 * Show tournament tourYear, date, course name, and course image.
 * TournamentCard also functions as link to each tournament's detail page.
 *
 * TournamentCard is rendered by TournamentList to show a "card" for each tournament.
 *
 * TournamentList -> TournamentCard
 *
 */

const TournamentCard = ({ date, courseName, imgUrl }) => {
  return (
    <div className="mb-5">
      <Link to={`/tournaments/${date}`} className="TournamentCard-link">
        <Card className="TournamentCard">
          {/* <Card.Title
            tag="h3"
            className=" display-6 bg-dark text-white mb-0 py-3"
          >
            {courseName}
          </Card.Title> */}

          <div className="TournamentCard-crop border-rounded">
            <img alt={`${courseName}`} src={imgUrl} />
          </div>
          <Card.Body className="py-3 bg-dark text-white">
            <h3 className="mb-0">
              {" "}
              {new Date(date).toLocaleDateString("en-US", {
                year: "numeric",
                month: "long",
                day: "numeric",
                timeZone: "UTC",
              })}
            </h3>
          </Card.Body>
        </Card>
      </Link>
    </div>
  );
};

export default TournamentCard;
