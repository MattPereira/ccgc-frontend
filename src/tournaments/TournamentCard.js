import React from "react";
import { Card } from "react-bootstrap";
import { Link } from "react-router-dom";
import "./TournamentCard.css";

/** Tournament card component.
 *
 * Show tournament tourYear, date, course name, and course image.
 * TournamentCard also functions as link to each tournament's detail page.
 *
 * TournamentCard is rendered by TournamentList to show a "card" for each tournament.
 *
 * TournamentList -> TournamentCard
 *
 */

const TournamentCard = ({ date, courseName, imgUrl, tourYears }) => {
  return (
    <div className="mb-5">
      <Link to={`/tournaments/${date}`} className="TournamentCard-link">
        <Card className="TournamentCard-card">
          <Card.Title tag="h3" className="bg-dark text-white mb-0 py-4">
            {courseName}
          </Card.Title>

          <div className="TournamentCard-crop">
            <img alt={`${courseName}`} src={imgUrl} />
          </div>

          <div className="bg-secondary text-white py-2">
            <h5 className="mb-0">
              {" "}
              {new Date(date).toLocaleDateString("en-US", {
                year: "numeric",
                month: "long",
                day: "numeric",
              })}
            </h5>
          </div>
        </Card>
      </Link>
    </div>
  );
};

export default TournamentCard;
