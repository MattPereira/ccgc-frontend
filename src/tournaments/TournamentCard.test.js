import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import TournamentCard from "./TournamentCard";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <TournamentCard
        date="2022-02-02"
        courseName="Pebble Beach Golf Course"
        imgUrl="/course-photos/pebble-beach"
      />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <TournamentCard
        date="2022-02-02"
        courseName="Pebble Beach Golf Course"
        imgUrl="/course-photos/pebble-beach"
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays tournament card", function () {
  render(
    <MemoryRouter>
      <TournamentCard
        date="2022-02-02"
        courseName="Pebble Beach Golf Course"
        imgUrl="/course-photos/pebble-beach"
      />
    </MemoryRouter>
  );

  const cardTitle = screen.getByText("Pebble Beach Golf Course");
  expect(cardTitle).toBeInTheDocument();

  const cardDate = screen.getByText("February 2, 2022");
  expect(cardDate).toBeInTheDocument();

  const cardImage = screen.getByAltText("Pebble Beach Golf Course");
  expect(cardImage).toBeInTheDocument();
});
