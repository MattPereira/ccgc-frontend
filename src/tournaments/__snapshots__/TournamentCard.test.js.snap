// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-5"
  >
    <a
      class="TournamentCard-link"
      href="/tournaments/2022-02-02"
    >
      <div
        class="TournamentCard-card card"
      >
        <div
          class=" display-6 bg-dark text-white mb-0 py-3 card-title h5"
          tag="h3"
        >
          Pebble Beach Golf Course
        </div>
        <div
          class="TournamentCard-crop"
        >
          <img
            alt="Pebble Beach Golf Course"
            src="/course-photos/pebble-beach"
          />
        </div>
        <div
          class="py-3 card-body"
        >
          <p
            class="mb-0 lead"
          >
             February 2, 2022
          </p>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
